api/v1alpha1/spire_server_config_types.go

```diff
@@ -165,10 +171,182 @@ type CASubject struct {
 	CommonName string `json:"commonName,omitempty"`
 }
 
+// UpstreamAuthority defines the configuration for the upstream certificate authority
+// that SPIRE Server will use to obtain its signing certificate.
+// It supports different plugins such as SPIRE, Vault, and cert-manager for upstream CA integration.
+type UpstreamAuthority struct {
+	// Type specifies the type of upstream authority plugin to use.
+	// Allowed values: "spire", "vault", "cert-manager".
+	// It determines which one of the optional configurations below should be populated.
+	Type string `json:"type,omitempty"`
+
+	// upstreamAuthoritySpire plugin uses credentials fetched from the Workload API to call an upstream
```

Comment:
nitpick: you may also need to update names in the API comments (e.g. upstreamAuthoritySpire -> spire) to keep consistency